library/dash/src/main/java/com/google/android/exoplayer2/source/dash/manifest/DashManifestParser.java

```diff
@@ -996,7 +1001,12 @@ protected EventMessage buildEvent(
       if (XmlPullParserUtil.isStartTag(xpp, "S")) {
         elapsedTime = parseLong(xpp, "t", elapsedTime);
         long duration = parseLong(xpp, "d", C.TIME_UNSET);
-        int count = 1 + parseInt(xpp, "r", 0);
+
+        //if repeat is -1 : length of each segment = duration / timescale and
+        //                   number of segments  = periodDuration / length of each segment
+        int repeat = parseInt(xpp,"r",0);
+        int count = repeat != -1? 1 + repeat : (int) (((periodDurationMs / 1000) * timescale) / duration);
```

Comment:
There is no mention in the DASH IOP that this value needs to be -1 specifically. It just says  "if the @r value is greater than or equal to zero" for one case and "else" for the other (=this) case. So can you check for < 0 instead?

Reply:
Re-submitted patch with the above change.